memgpt/local_llm/llm_chat_completion_wrappers/configurable_wrapper.py

```diff
@@ -0,0 +1,301 @@
+import json
+from typing import List
+
+import yaml
+
+from .wrapper_base import LLMChatCompletionWrapper
+from ...constants import JSON_ENSURE_ASCII
+from ..json_parser import clean_json
+from ...constants import JSON_ENSURE_ASCII
+from ...errors import LLMJSONParsingError
+
+
+# A configurable model agnostic wrapper.
+class ConfigurableWrapper(LLMChatCompletionWrapper):
```

Comment:
Maybe we should call this `ConfigurableJSONOutputWrapper`?

(Since it's specific to JSON outputs, vs a potential YAML version)

Reply:
Sure.

Reply:
Do you think I need to add other configurable parameters, like first message?

Reply:
Maybe not needed for this initial PR? We can maybe add all that in a follow-up PR that obsoletes the old wrapper classes and replaces them with this the configurable wrapper (assuming it fully supports all the existing ones).

Reply:
I will add it now. Will be an easy change.